code/iaas/allocator/src/main/java/io/cattle/platform/allocator/constraint/NetworkContainerConstraintProvider.java

```diff
@@ -0,0 +1,46 @@
+package io.cattle.platform.allocator.constraint;
+
+import io.cattle.platform.allocator.service.AllocationAttempt;
+import io.cattle.platform.allocator.service.AllocationLog;
+import io.cattle.platform.core.dao.GenericMapDao;
+import io.cattle.platform.core.model.Instance;
+import io.cattle.platform.docker.constants.DockerInstanceConstants;
+import io.cattle.platform.docker.constants.DockerNetworkConstants;
+import io.cattle.platform.json.JsonMapper;
+import io.cattle.platform.object.util.DataAccessor;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import javax.inject.Inject;
+
+public class NetworkContainerConstraintProvider implements AllocationConstraintsProvider {
+    @Inject
+    GenericMapDao mapDao;
+
+    @Inject
+    JsonMapper jsonMapper;
+
+    @Override
+    public void appendConstraints(AllocationAttempt attempt, AllocationLog log,
+            List<Constraint> constraints) {
+        Instance instance = attempt.getInstance();
+
+        if (instance != null) {
+            String networkMode = DataAccessor.fields(instance).withKey(DockerInstanceConstants.FIELD_NETWORK_MODE).as(String.class);
+            if (DockerNetworkConstants.NETWORK_MODE_CONTAINER.equals(networkMode)) {
+                Integer containerId = instance.getNetworkContainerId().intValue();
```

Comment:
In the current test we incorrectly set container network mode w/o setting the networkContainerId.  This test is fixed in https://github.com/rancherio/cattle/pull/578/files.  But this bad test does cause a NPE here.  I would say that if networkContainerId is null, we just don't apply the CollcationConstraint.